### PR TITLE
feat: s3, config changes, metrics

### PR DESCRIPTION
**Pull request by deepankarm**

### CODE REVIEW

**Comment by deepankarm** [comment]
@jacobowitz I haven't tested `download_from_s3`, please test & edit according to your requirements.

**Comment by jacobowitz** [comment]
@deepankarm 
I made some minor adjustments to the S3 download functionality. I tried it with some small test file and it works as expected. Thank you :)

Is it somehow possible to move this aws module somewhere so that we can access it also in https://github.com/jina-ai/examples
We will need it there as well for dataset access, e.g. these prs:
https://github.com/jina-ai/examples/pull/459
https://github.com/jina-ai/examples/pull/477
https://github.com/jina-ai/examples/pull/460

File: distributed/aws/s3.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                    self._client.upload_file(filename, self._bucket, f'{key}/{filename}')
+        except Exception as exp:
+            logger.exception(f'Got following exception while uploading object to S3 \n{exp}')
+
+    def get(self, key: str, local_path: str = '.'):
+        try:
+            logger.info(f'Downloading object from `{self._bucket}:{key}` to local directory: {local_path}')
+            # for filename in glob.iglob(path + '**/**', recursive=True):
+            self._client.download_file(self._bucket, key, local_path)
+        except Exception as exp:
```
Comment (jacobowitz) [inline distributed/aws/s3.py:34]: We may want to raise those to the caller instead as it is unclear here what to do in these cases?

**Review by jacobowitz — COMMENTED** [review]
Only had one minor comment and the general question regarding where to put aws module.

Otherwise this PR looks good to me, thx for the work!

File: distributed/aws/s3.py (hunk 1 above)
Comment (deepankarm) [reply distributed/aws/s3.py:34]: Good point. Done

**Comment by deepankarm** [comment]
> Only had one minor comment and the general question regarding where to put aws module.

I [wrote this not so efficient code](https://github.com/jina-ai/cloud-ops/tree/master/aws) a while back. We can probably keep this in a separate repo & add that as a pip+git requirement? That way we can use it with different codebase.

The concern would be the users of examples repo wouldn't have access to jina's aws account, hence wouldn't be able to fetch from private s3 bucket.

**Pull request merged**